### PR TITLE
PCA analysis page

### PR DESCRIPTION
**Pull request by waxlamp**
TODO
- [x] I need some help from @jtomeck to put a spacer between the plot names and their switches in the control panel

### CODE REVIEW

File: web/src/components/vis/PcaPage/LoadingsPlot.vue [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    showCrosshairs: {
+      type: Boolean,
+      required: true,
+    },
+  },
+};
+</script>
+
+<template lang="pug">
+vis-tile(title="PCA Loadings Plot", :loading="plot.loading", svg-download, v-if="plot")
```
Comment (sgratzl) [inline web/src/components/vis/PcaPage/LoadingsPlot.vue]: what happened to the help texts?

File: web/src/components/vis/PcaPage/ScorePlot.vue [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+      :pc-coords="pcCoords",
+      :row-labels="rowLabels",
+      :colors="dataset.groupLevels",
+      :group-labels="groupLabels",
+      :eigenvalues="eigenvalues",
+      :columns="columns",
+      :pc-x="pcX",
+      :pc-y="pcY",
+      :show-ellipses="showEllipses")
+div(v-else)
```
Comment (subdavis) [inline web/src/components/vis/PcaPage/ScorePlot.vue]: I'm not sure this loader belongs at this level in the tree.  If `plot` is not defined, the dataset has yet to load, so you'll have 3 graphs with spinners that are all showing the same load event.

Might be appropriate to bump this up to the parent and not show the grid at all until the dataset is loaded.   `plot` is really just a proxy for `getters.ready`.

File: web/src/components/vis/PcaPage/PcaPage.vue [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+      v-card-actions
+        v-layout(column)
+          v-text-field.py-2(
+            :disabled="!showScree",
+            hide-details,
+            type="number",
+            label="Number of PCs",
+            min="1",
+            outline,
+            v-model="numComponentsVal")
```
Comment (sgratzl) [inline web/src/components/vis/PcaPage/PcaPage.vue]: missing indentation of two space to be valid

File: web/src/components/vis/PcaPage/PcaPage.vue [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+          v-switch.ma-0.py-2(
+            v-model="showCutoffs",
+            label="Cutoff lines",
+            :disabled="!showScree",
+            hide-details)
```
Comment (sgratzl) [inline web/src/components/vis/PcaPage/PcaPage.vue]: missing indentation of two space to be valid

File: web/src/components/vis/PcaPage/PcaPage.vue (hunk 3 above)
Comment (waxlamp) [reply web/src/components/vis/PcaPage/PcaPage.vue]: c9cafed should fix this and the other lint problem.

File: web/src/components/vis/PcaPage/PcaPage.vue (hunk 4 above)
Comment (waxlamp) [reply web/src/components/vis/PcaPage/PcaPage.vue]: c9cafed should fix this and the other lint problem.

File: web/src/components/vis/PcaPage/ScorePlot.vue (hunk 2 above)
Comment (waxlamp) [reply web/src/components/vis/PcaPage/ScorePlot.vue]: I promoted this check up to the page level, using the store directly to check for data readiness. Check out my approach in 29ca24d and let me know if that looks good.

File: web/src/components/vis/PcaPage/LoadingsPlot.vue (hunk 1 above)
Comment (waxlamp) [reply web/src/components/vis/PcaPage/LoadingsPlot.vue]: Added help panels back in via 51f80a4 and 56a9d41.

**Pull request merged**